File: svws-core/src/main/java/de/svws_nrw/core/data/schule/Kindergarten.java
package de.svws_nrw.core.data.schule;

import de.svws_nrw.transpiler.TranspilerDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Die Klasse beschreibt den schulspezifischen Eintrag im Katalog der Kindergärten.
 */
@XmlRootElement
@Schema(description = "ein Eintrag in dem Katalog der schulspezifischen Kindergärten.")
@TranspilerDTO
public class Kindergarten {

	/** Die ID des Kindergartens. */
	@Schema(description = "Die ID des Kindergartens.", example = "4711", accessMode = Schema.AccessMode.READ_ONLY)
	public long id;

	/** Die Bezeichnung des Kindergartens. */
	@Schema(description = "Die Bezeichnung des Kindergartens.", example = "Kita Sonnenschein")
	public @NotNull String bezeichnung = "";

	/** Bemerkung zum Kindergartens. */
	@Schema(description = "Die Bemerkung zum Kindergartens.", example = "Ist geschlossen")
	public String bemerkung;

	/** Die Telefonnummer des Kindergartens. */
	@Schema(description = "Die Telefonnummer des Kindergartens.", example = "00007-4711")
	public String tel;

	/** Die E-Mail des Kindergartens. */
	@Schema(description = "Die E-Mail des Kindergartens.", example = "kita@sonnenschein.de")
	public String email;

	/** Der Strassenname des Kindergartens. */
	@Schema(description = "Der Strassenname des Kindergartens.", example = "Musterweg")
	public String strassenname;

	/** Die Hausnummer des Kindergartens. */
	@Schema(description = "Die Hausnummer des Kindergartens.", example = "12")
	public String hausNr;

	/** der Hausnummerzusatz des Kindergartens. */
	@Schema(description = "Der Hausnummerzusatz des Kindergartens.", example = "a-d")
	public String hausNrZusatz;

	/** Die PLZ des Kindergartens. */
	@Schema(description = "Die PLZ des Kindergartens.", example = "42287")
	public String plz;

	/** Der Ort des Kindergartens. */
	@Schema(description = "Der Ort des Kindergartens.", example = "Düsseldorf")
	public String ort;

	/** Gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an. */
	@Schema(description = "Gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an.", example = "1")
	public int sortierung;

	/** Gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht. */
	@Schema(description = "Gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht.", example = "true")
	public boolean istSichtbar;

	/** Gibt an, ob der Kindergarten in anderen Datenbanktabellen referenziert ist oder nicht. */
	@Schema(description = "Gibt an, ob der Kindergarten in anderen Datenbanktabellen referenziert ist oder nicht.",
			example = "true",
			accessMode = Schema.AccessMode.READ_ONLY)
	public boolean referenziertInAnderenTabellen;

}

File: svws-webclient/core/src/core/data/schule/Kindergarten.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { Class } from '../../../java/lang/Class';

export class Kindergarten extends JavaObject {

	/**
	 * Die ID des Kindergartens.
	 */
	public id: number = 0;

	/**
	 * Die Bezeichnung des Kindergartens.
	 */
	public bezeichnung: string = "";

	/**
	 * Bemerkung zum Kindergartens.
	 */
	public bemerkung: string | null = null;

	/**
	 * Die Telefonnummer des Kindergartens.
	 */
	public tel: string | null = null;

	/**
	 * Die E-Mail des Kindergartens.
	 */
	public email: string | null = null;

	/**
	 * Der Strassenname des Kindergartens.
	 */
	public strassenname: string | null = null;

	/**
	 * Die Hausnummer des Kindergartens.
	 */
	public hausNr: string | null = null;

	/**
	 * der Hausnummerzusatz des Kindergartens.
	 */
	public hausNrZusatz: string | null = null;

	/**
	 * Die PLZ des Kindergartens.
	 */
	public plz: string | null = null;

	/**
	 * Der Ort des Kindergartens.
	 */
	public ort: string | null = null;

	/**
	 * Gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an.
	 */
	public sortierung: number = 0;

	/**
	 * Gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht.
	 */
	public istSichtbar: boolean = false;

	/**
	 * Gibt an, ob der Kindergarten in anderen Datenbanktabellen referenziert ist oder nicht.
	 */
	public referenziertInAnderenTabellen: boolean = false;


	public constructor() {
		super();
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.data.schule.Kindergarten';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.core.data.schule.Kindergarten'].includes(name);
	}

	public static readonly class = new Class<Kindergarten>('de.svws_nrw.core.data.schule.Kindergarten');

	public static transpilerFromJSON(json: string): Kindergarten {
		const obj = JSON.parse(json) as Partial<Kindergarten>;
		const result = new Kindergarten();
		if (obj.id === undefined)
			throw new Error('invalid json format, missing attribute id');
		result.id = obj.id;
		if (obj.bezeichnung === undefined)
			throw new Error('invalid json format, missing attribute bezeichnung');
		result.bezeichnung = obj.bezeichnung;
		result.bemerkung = (obj.bemerkung === undefined) ? null : obj.bemerkung === null ? null : obj.bemerkung;
		result.tel = (obj.tel === undefined) ? null : obj.tel === null ? null : obj.tel;
		result.email = (obj.email === undefined) ? null : obj.email === null ? null : obj.email;
		result.strassenname = (obj.strassenname === undefined) ? null : obj.strassenname === null ? null : obj.strassenname;
		result.hausNr = (obj.hausNr === undefined) ? null : obj.hausNr === null ? null : obj.hausNr;
		result.hausNrZusatz = (obj.hausNrZusatz === undefined) ? null : obj.hausNrZusatz === null ? null : obj.hausNrZusatz;
		result.plz = (obj.plz === undefined) ? null : obj.plz === null ? null : obj.plz;
		result.ort = (obj.ort === undefined) ? null : obj.ort === null ? null : obj.ort;
		if (obj.sortierung === undefined)
			throw new Error('invalid json format, missing attribute sortierung');
		result.sortierung = obj.sortierung;
		if (obj.istSichtbar === undefined)
			throw new Error('invalid json format, missing attribute istSichtbar');
		result.istSichtbar = obj.istSichtbar;
		if (obj.referenziertInAnderenTabellen === undefined)
			throw new Error('invalid json format, missing attribute referenziertInAnderenTabellen');
		result.referenziertInAnderenTabellen = obj.referenziertInAnderenTabellen;
		return result;
	}

	public static transpilerToJSON(obj: Kindergarten): string {
		let result = '{';
		result += '"id" : ' + obj.id.toString() + ',';
		result += '"bezeichnung" : ' + JSON.stringify(obj.bezeichnung) + ',';
		result += '"bemerkung" : ' + ((obj.bemerkung === null) ? 'null' : JSON.stringify(obj.bemerkung)) + ',';
		result += '"tel" : ' + ((obj.tel === null) ? 'null' : JSON.stringify(obj.tel)) + ',';
		result += '"email" : ' + ((obj.email === null) ? 'null' : JSON.stringify(obj.email)) + ',';
		result += '"strassenname" : ' + ((obj.strassenname === null) ? 'null' : JSON.stringify(obj.strassenname)) + ',';
		result += '"hausNr" : ' + ((obj.hausNr === null) ? 'null' : JSON.stringify(obj.hausNr)) + ',';
		result += '"hausNrZusatz" : ' + ((obj.hausNrZusatz === null) ? 'null' : JSON.stringify(obj.hausNrZusatz)) + ',';
		result += '"plz" : ' + ((obj.plz === null) ? 'null' : JSON.stringify(obj.plz)) + ',';
		result += '"ort" : ' + ((obj.ort === null) ? 'null' : JSON.stringify(obj.ort)) + ',';
		result += '"sortierung" : ' + obj.sortierung.toString() + ',';
		result += '"istSichtbar" : ' + obj.istSichtbar.toString() + ',';
		result += '"referenziertInAnderenTabellen" : ' + obj.referenziertInAnderenTabellen.toString() + ',';
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

	public static transpilerToJSONPatch(obj: Partial<Kindergarten>): string {
		let result = '{';
		if (obj.id !== undefined) {
			result += '"id" : ' + obj.id.toString() + ',';
		}
		if (obj.bezeichnung !== undefined) {
			result += '"bezeichnung" : ' + JSON.stringify(obj.bezeichnung) + ',';
		}
		if (obj.bemerkung !== undefined) {
			result += '"bemerkung" : ' + ((obj.bemerkung === null) ? 'null' : JSON.stringify(obj.bemerkung)) + ',';
		}
		if (obj.tel !== undefined) {
			result += '"tel" : ' + ((obj.tel === null) ? 'null' : JSON.stringify(obj.tel)) + ',';
		}
		if (obj.email !== undefined) {
			result += '"email" : ' + ((obj.email === null) ? 'null' : JSON.stringify(obj.email)) + ',';
		}
		if (obj.strassenname !== undefined) {
			result += '"strassenname" : ' + ((obj.strassenname === null) ? 'null' : JSON.stringify(obj.strassenname)) + ',';
		}
		if (obj.hausNr !== undefined) {
			result += '"hausNr" : ' + ((obj.hausNr === null) ? 'null' : JSON.stringify(obj.hausNr)) + ',';
		}
		if (obj.hausNrZusatz !== undefined) {
			result += '"hausNrZusatz" : ' + ((obj.hausNrZusatz === null) ? 'null' : JSON.stringify(obj.hausNrZusatz)) + ',';
		}
		if (obj.plz !== undefined) {
			result += '"plz" : ' + ((obj.plz === null) ? 'null' : JSON.stringify(obj.plz)) + ',';
		}
		if (obj.ort !== undefined) {
			result += '"ort" : ' + ((obj.ort === null) ? 'null' : JSON.stringify(obj.ort)) + ',';
		}
		if (obj.sortierung !== undefined) {
			result += '"sortierung" : ' + obj.sortierung.toString() + ',';
		}
		if (obj.istSichtbar !== undefined) {
			result += '"istSichtbar" : ' + obj.istSichtbar.toString() + ',';
		}
		if (obj.referenziertInAnderenTabellen !== undefined) {
			result += '"referenziertInAnderenTabellen" : ' + obj.referenziertInAnderenTabellen.toString() + ',';
		}
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

}

export function cast_de_svws_nrw_core_data_schule_Kindergarten(obj: unknown): Kindergarten {
	return obj as Kindergarten;
}
